docs/api-ref.md

```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
+   TSC.Permission.Capability.Read: TSC.Permission.Mode.Allow
+ }
+
+ project_rules = TSC.PermissionsRule(
+  grantee=project_item,
+  capabilities=project_capabilities
+ )
+ 
+ server.projects.update_permission(project_item, [project_rules])
+``` 
+
+
+#### projects.delete_permission
+```py
+projects.delete_permission(item, rules)
+```
+
+Delete project permissions.
+
+REST API: [Delete Project Permissions](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_permissions.htm#delete_project_permission){:target="_blank"}
+
+**Parameters**
+
+Name  |  Description  
+:--- | :---  
+`item`   | A project object.
+`rules`   | A list of PermissionsRule objects or a single PermissionsRule object.
+
+**Example**  
+ 
+```py
+# import tableauserverclient as TSC  
+# server = TSC.Server('https://MY-SERVER')  
+# sign in, etc.  
+
+ all_project_items, pagination_item = server.projects.get()
+ project_item = all_project_items[0]
+ 
+ project_capabilities = {
+   TSC.Permission.Capability.Read: TSC.Permission.Mode.Allow
+ }
+
+ project_rules = TSC.PermissionsRule(
+  grantee=project_item,
+  capabilities=project_capabilities
+ )
+ 
+ server.projects.delete_permission(project_item)
+``` 
+
+
+#### projects.populate_workbook_default_permissions
+```py
+projects.populate_workbook_default_permissions(item)
+```
+
+Queries the default workbook permissions, parses and stores the returned the permissions.
+
+REST API: [Query Default Permissions](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_permissions.htm#query_default_permissions){:target="_blank"}  
+
+**Parameters**
+
+Name  |  Description  
+:--- | :---  
+`item`   | A project object.
+
+**Example**  
+ 
+```py
+# import tableauserverclient as TSC  
+# server = TSC.Server('https://MY-SERVER')  
+# sign in, etc.  
+
+ all_project_items, pagination_item = server.projects.get()
+ project_item = all_project_items[0]
+ server.projects.populate_workbook_default_permissions(project_item)
+``` 
+
+#### projects.populate_datasource_default_permissions
+```py
+projects.populate_datasource_default_permissions(item)
+```
+
+Queries the default datasource permissions, parses and stores the returned the permissions.
+
+REST API: [Query Default Permissions](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_permissions.htm#query_default_permissions){:target="_blank"}  
+
+**Parameters**
+
+Name  |  Description  
+:--- | :---  
+`item`   | A project object.
+
+**Example**  
+ 
+```py
+# import tableauserverclient as TSC  
+# server = TSC.Server('https://MY-SERVER')  
+# sign in, etc.  
+
+ all_project_items, pagination_item = server.projects.get()
+ project_item = all_project_items[0]
+ server.projects.populate_datasource_default_permissions(project_item)
+``` 
+
+
+#### projects.populate_flow_default_permissions
+```py
+projects.populate_flow_default_permissions(item)
+```
+
+Queries the default flow permissions, parses and stores the returned the permissions.
```

Comment:
Again:

Queries the default permissions for flows in a project, parses and stores the returned the permissions.